Shared/Realm.Sync.Shared/SyncConfiguration.cs

```diff
@@ -0,0 +1,55 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Realms.Sync
+{
+    public class SyncConfiguration : RealmConfiguration
+    {
+        public Uri ServerUri { get; private set; }
+
+        public User User { get; private set; }
+
+        public bool ShouldDeleteRealmOnLogOut { get; private set; }
+
+        public SyncConfiguration(User user, Uri serverUri)
```

Comment:
That's what I dislike ;) In almost all cases when an API requires `Uri`, I'll pass in `new Uri(myStringUri)`, which feels annoying and cumbersome. We can always do an `Uri.TryParse` and if it's not well formed throw an exception to let the user know something's wrong.